homotopy-model/src/proof.rs

```diff
@@ -165,6 +169,7 @@ impl Action {
             Self::ClearWorkspace => proof.workspace.is_some(),
             Self::ClearBoundary => proof.boundary.is_some(),
             Self::SelectGenerator(_) => true,
+            Self::Strictify(g) => g.dimension > 0,
```

Comment:
This should also return false if the source = target (basically the rule is `Action::is_valid` should return false whenever the action handler returns false).